railgun/test/integration/knative_test.go

```diff
@@ -0,0 +1,286 @@
+//+build integration_tests
+
+package integration
+
+import (
+	"crypto/tls"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"time"
+
+	"bytes"
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	types "k8s.io/apimachinery/pkg/types"
+
+	"github.com/kong/kubernetes-testing-framework/pkg/kind"
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	knativeCrds = "https://github.com/knative/serving/releases/download/v0.13.0/serving-crds.yaml"
+	knativeCore = "https://github.com/knative/serving/releases/download/v0.13.0/serving-core.yaml"
+)
+
+func isKnativeReady(ctx context.Context, cluster kind.Cluster) bool {
+	timeout := 1
+	for timeout < 60 {
```

Comment:
Using Golang's native `time` library you can create a timeout that doesn't require you to do do specific 1 second sleep intervals, e.g.:

```
timeout := time.Now().Add(time.Minute * 1)
for timeout.After(time.Now()) {
   fmt.Println("doing work!")
   time.Sleep(time.Millisecond * 50)
}
```

The folowing docs may be helpful in this regard and include some examples:

- https://pkg.go.dev/time#Time.After
- https://pkg.go.dev/time#Time.Add

Reply:
Your suggestion of assert.Eventually is more concise solution in this case. 
What do you think ?

Reply:
Yes, I wrote this before I had thought on this. assert/require Eventually will both do the correct thing.